src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1818,34 +1891,111 @@ private static void validateStyleModules(Set<Node> checks, Set<Node> configs,
+
+            if (!moduleIsConfig && !moduleIsCheck) {
+                skip = true;
+                continue;
+            }
+
+            assertWithMessage(styleName + "_style.xml rule '" + ruleName + "' module '"
+                    + moduleName + "' shouldn't end with 'Check'")
+                    .that(moduleName.endsWith("Check"))
+                    .isFalse();
+
+            styleChecks.remove(moduleName);
+
+            Node config = null;
+
+            try {
+                config = itrConfigs.next();
+            }
+            catch (NoSuchElementException ignore) {
+                assertWithMessage(styleName + "_style.xml rule '" + ruleName + "' module '"
+                        + moduleName + "' is missing the config link: config").fail();
+            }
+
+            final String configUrl = config.getAttributes().getNamedItem("href")
+                    .getTextContent();
+
+            final String expectedUrl = "https://github.com/search?q="
+                    + "path%3Asrc%2Fmain%2Fresources%20path%3A**%2F" + styleName
+                    + "_checks.xml+repo%3Acheckstyle%2Fcheckstyle+" + moduleName;
+
+            assertWithMessage("google_style.xml rule '" + ruleName + "' module '"
+                    + moduleName + "' should have matching config url")
+                    .that(configUrl)
+                    .isEqualTo(expectedUrl);
+
+        }
+
+        if (!skip) {
+            if (itrConfigs.hasNext()) {
+                final Node config = itrConfigs.next();
+                final String configUrl = config.getAttributes().getNamedItem("href")
+                        .getTextContent();
+                final String[] parts = ruleName.split(" ", 2);
+                final String extractedRuleName = parts[1].trim().replaceAll(" ", "");
+
+                assertWithMessage("google_style.xml rule '" + ruleName + "' rule '"
+                        + "' should have matching test url")
+                        .that(configUrl)
+                        .startsWith("https://github.com/checkstyle/checkstyle/"
+                                + "blob/master/src/it/java/com/google"
+                                + "/checkstyle/test/");
+
+                assertWithMessage("google_style.xml rule '" + ruleName
+                        + "' should have matching test url")
+                        .that(configUrl)
+                        .endsWith("/" + extractedRuleName + "Test.java");
+
+                assertWithMessage("google_style.xml rule '" + ruleName
+                        + "' should have a test that exists")
+                        .that(new File(configUrl.substring(53).replace('/',
+                                File.separatorChar)).exists())
+                        .isTrue();
+            }
+            else {
+                assertWithMessage("google_style.xml rule '" + ruleName + "' should have a test url")
+                        .fail();
```

Comment:
Change this to assert `hasChecks` is `false`.

Reply:
done